src/sonic_ax_impl/lib/perseverantsocket.py

```diff
@@ -0,0 +1,35 @@
+import socket
+
+class PerseverantSocket:
+    def __init__(self, family=socket.AF_INET, type=socket.SOCK_STREAM, proto=0, fileno=None, address_tuple=None, *args, **kwargs):
+        self._connected = False
+        self.address_tuple = address_tuple
+        self.family = family
+        self.type = type
```

Comment:
type is already used by a standard python library.

Reply:
I am following parameter naming in [https://docs.python.org/3/library/socket.html#socket.socket](https://docs.python.org/3/library/socket.html#socket.socket). Since it is safe, let's keep it.

---
In reply to: [355809292](https://github.com/Azure/sonic-snmpagent/pull/119#discussion_r355809292) [](ancestors = 355809292)